File: 02. Algorithms/04. Recursion/Problems/01. 509. Fibonacci Number/fibonacci.py

## Finds the Nth number in the Fibonacci Series
def fib(n: int) -> int:

    if n <= 1:  # returns 0 if fib(0), and 1 if fib(1)
        return n

    return fib(n-1) + fib(n-2)

print(fib(12))




## Generates a Fibonacci Series with two starting numbers, up until the Nth number.
def fib_sequence(n: int, x: int = 0,y: int = 1) -> list:

    if n == 0:
        return [y]

    return [y]+fib_sequence(n-1,y,x+y)

print(fib_sequence(15,1,1))

File: 02. Algorithms/04. Recursion/Problems/01. 509. Fibonacci Number/fibonacci.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    
    const dfs = (n) => {
        if(n <= 0) {
            return 0;
        }
        if(n === 1) {
            return 1;
        }
        return dfs(n-1) + dfs(n-2);
    }

    return dfs(n);
};
